apps/pipeline/lib/pipeline/writer/table_writer/compaction.ex

```diff
@@ -66,4 +67,17 @@ defmodule Pipeline.Writer.TableWriter.Compaction do
       results.rows
     end
   end
+
+  def add_dataset_record_event_count(table_name, old_row_count, new_row_count) do
+    if !String.ends_with?(table_name, "__json") do
```

Comment:
The convention we use is to do this sort of logic via pattern matching in the function head.